fractal_tasks_core/tasks/_zarr_utils.py

```diff
@@ -102,3 +105,65 @@ def _split_well_path_image_path(zarr_url: str) -> tuple[str, str]:
     well_path = "/".join(zarr_url.split("/")[:-1])
     img_path = zarr_url.split("/")[-1]
     return well_path, img_path
+
+
+def _split_base_suffix(input: str) -> tuple[str, str]:
+    parts = input.split("_")
+    base = parts[0]
+    if len(parts) > 1:
+        suffix = "_".join(parts[1:])
+    else:
+        suffix = ""
+    return base, suffix
+
+
+def _get_matching_ref_cycle_path_heuristic(
+    path_list: list[str], path: str
+) -> str:
+    """
+    Pick the best match from path_list to a given path
+
+    This is a workaround to find the reference registration cycle when there
+    are multiple OME-Zarrs with the same acquisition identifier in the well
+    metadata and we need to find which one is the reference for a given path.
+
+    Args:
+        path_list: List of paths to OME-Zarr images in the well metadata. For
+            example: ['0', '0_illum_corr']
+        path: A given path for which we want to find the reference image. For
+            example, '1_illum_corr'
+
+    Returns:
+        The best matching reference path. If no direct match is found, it
+        returns the most similar one based on suffix hierarchy or the base
+        path if applicable. For example, '0_illum_corr' with the example
+        inputs above.
+    """
+    # Extract the base number and suffix from the input path
+    base, suffix = _split_base_suffix(path)
+
+    # Iterate over each path in the list to find the best match with the same
+    # base
+    for p in path_list:
+        # Split the list path into base and suffix
+        p_base, p_suffix = _split_base_suffix(p)
+
+        # If suffices match, it's the match.
+        if p_suffix == suffix:
+            return p
+
+        # Check for matching base
+        if p_base == base:
+            # If no suffix in input, the base path is the best match
+            if not suffix and not p_suffix:
+                return p
+            # If suffix matches, it's a definitive best match
+            if suffix == p_suffix:
```

Comment:
This can never be `True`, because we already returned `p` when `p_suffix == suffix`